arch/risc-v/src/common/crt0.c

```diff
@@ -88,6 +88,71 @@ static void sig_trampoline(void)
   );
 }
 
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#if defined(__cplusplus)
+extern "C"
+{
+#endif
+
+/* Linker defined symbols to .ctors and .dtors */
+
+  extern void (*_sctors)(void);
+  extern void (*_ectors)(void);
+  extern void (*_sdtors)(void);
+  extern void (*_edtors)(void);
```

Comment:
```suggestion
extern void (*_sctors)(void);
extern void (*_ectors)(void);
extern void (*_sdtors)(void);
extern void (*_edtors)(void);
```

Reply:
I did it this way because the code style check complains if they are not indented. Other similar (c++) files do it this way too.

Reply:
Ok. I didn't know that. So maybe we can just use `EXTERN` instead of `extern`?

Reply:
The `EXTERN` is undefined at the end of each header file... So maybe we can add
```
#if defined(__cplusplus)
#define EXTERN extern "C"
#else
#define EXTERN extern
#endif
```
on top of this file

Reply:
Or move those externs to `riscv_internal.h` and place somewhere near
```
EXTERN uint32_t _stext;           /* Start of .text */
EXTERN uint32_t _etext;           /* End_1 of .text + .rodata */
EXTERN const uint32_t _eronly;    /* End+1 of read only section (.text + .rodata) */
EXTERN uint32_t _sdata;           /* Start of .data */
EXTERN uint32_t _edata;           /* End+1 of .data */
EXTERN uint32_t _sbss;            /* Start of .bss */
EXTERN uint32_t _ebss;            /* End+1 of .bss */
EXTERN uint32_t _stdata;          /* Start of .tdata */
EXTERN uint32_t _etdata;          /* End+1 of .tdata */
EXTERN uint32_t _stbss;           /* Start of .tbss */
EXTERN uint32_t _etbss;           /* End+1 of .tbss */
```

Reply:
I guess this will work as well, yes. I can do it like this but it goes to next week. For the ctors/dtors I don't think it will help, they will still have the funny indent. Unless we add
```
extern "C"
{
```
Which might still cause the code style check to produce an error